packages/core/src/utils.ts

```diff
@@ -83,6 +84,13 @@ export function shortenEns(ens: string): string {
   return ens.length > 11 ? `${ens.slice(0, 4)}...${ens.slice(-6)}` : ens
 }
 
+export function weiToEth(wei: any): string {
+  const bigWei = new BigNumber(wei._hex)
+  const bigWeiDiv = bigWei.dividedBy(1e18)
+  const formattedWei = bigWei.dividedBy(1e18).toFixed() === '0' ? bigWeiDiv.toFixed(1) : bigWeiDiv.toFixed()
+  return formattedWei
+}
```

Comment:
```suggestion
export function weiToEth(wei: any): string {
  return new BigNumber(wei).div(1e18).toString()
}
```

Reply:
Not sure what the `_hex` refers to here. This function should take a string value in `wei` as input and return the string in `ether` without the full decimal places. Then where ever it is being rendered, it can be decided how many decimals render.

Reply:
logging wei will give you `BigNumber {_hex: '0x05443223fb7515db', _isBigNumber: true}`. Returning `new BigNumber(wei).div(1e18).toString()` without accessing _hex will give you `NaN`.

Reply:
That would mean that the function is being called with a Bignumber instance rather than a string. My suggestion would be if possible, to ensure the function is always called with a string value for consistency.